### PR TITLE
Restructure methods

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Restructures the space weather methods so that there is one method for each data type.  Also updates the meta data to conform with the soon-to-be released structure: https://github.com/pysat/pysat/pull/587.

Addresses #21 

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

This has been tested by running the unit tests locally.

## Test Configuration
* Operating system: OS X Mojave
* Version number: Python 3.7
* Any details about your local setup that are relevant: working with pysat on the PR listed in description.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

**This PR should not be merged until the tagged pysat PR is merged**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [0.0.2] - 2020-XX-XX
 - Added real-time ACE instrument
-- Fixed bugs introduced by changes in pysat-3.0
+- Fixed bugs and restructured code to comply with changes in pysat-3.0
 - Added user documentation
 - Simplified setup and testing environments
+- Separated space seather methods into sub-modules by instrument
```
Comment (rstoneback) [inline CHANGELOG.md:10]:
```suggestion
- Separated space weather methods into sub-modules by instrument
```

File: pysatSpaceWeather/instruments/methods/dst.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    ----------
+    name : string
+        Name of space weather index, eg, dst, f107, kp
+    tag : string
+        Tag of the space waether index
+
+    """
+
+    refs = {'dst': {'': ''.join([
+        'see referenece list and publication at: Sugiura M. and T. Kamei, '
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/methods/dst.py:37]:
```suggestion
        'See referenece list and publication at: Sugiura M. and T. Kamei, '
```

File: pysatSpaceWeather/instruments/methods/kp_ap.py [hunk 3]
```diff
@@ -47,20 +46,344 @@ def references(name, tag):
...
+        kp_inst.load(date=inst.date, verifyPad=True)
+
+    # Begin filtering, starting at the beginning of the instrument data
+    sel_data = kp_inst[(inst.date - pds.DateOffset(days=1)):
+                       (inst.date + pds.DateOffset(days=1))]
+    ind, = np.where((sel_data['Kp'] > max_kp) | (sel_data['Kp'] < min_kp))
+    for lind in ind:
+        sind = sel_data.index[lind]
+        eind = sind + pds.DateOffset(hours=filter_time)
+        inst.data[sind:eind] = np.nan
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/methods/kp_ap.py:377]: Going through the Instrument itself provides (in theory!) the most data compatible interface.

```suggestion
        inst[sind:eind] = np.nan
```

**Review by rstoneback — COMMENTED** [review]
Thanks @aburrell! Took a look at filter_geomag per request and it looks good. I went ahead and took a look at other changes. Found a couple small typos. I'd hit approve except I may be a bit early again.

**Comment by aburrell** [comment]
Ok @rstoneback I think it's ready now!

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 - Fixed bugs and restructured code to comply with changes in pysat-3.0
 - Added user documentation
 - Simplified setup and testing environments
-- Separated space seather methods into sub-modules by instrument
+- Separated space weather methods into sub-modules by instrument
```
Comment (rstoneback) [inline CHANGELOG.md:10]: 👍

**Review by rstoneback — APPROVED** [review]
Thanks @aburrell! New changes look good. Passes tests locally. I double-checked for residual `default` methods. Didn't find any.

**Pull request merged**